packages/@aws-cdk/cloudformation-include/lib/cfn-include.ts

```diff
@@ -0,0 +1,128 @@
+   * as they did in the original CloudFormation template file.
+   *
+   * **Note**: regardless of whether this option is true or false,
+   * the {@link CfnInclude.getResource} method always uses the original logical ID of the resource,
+   * as specified in the template file.
+   *
+   * @default true
+   */
+  // readonly preserveLogicalIds?: boolean;
+}
+
+/**
+ * A construct that allows you to import an existing CloudFormation template file into your CDK application.
+ * You can retrieve any of the resources defined in the template file by calling the {@link getResource} method.
+ * Any modifications you make on the returned resource objects will be reflected in the resulting CDK template.
+ *
+ * @see getResource
+ */
+export class CfnInclude extends core.CfnElement {
+  private readonly resources: { [logicalId: string]: core.CfnResource } = {};
+  private readonly template: any;
+  private readonly preserveLogicalIds: boolean;
+
+  constructor(scope: core.Construct, id: string, props: CfnIncludeProps) {
+    super(scope, id);
+
+    // read the template into a JS object
+    this.template = futils.readJsonSync(props.templateFile);
+
+    // ToDo implement preserveLogicalIds=false
+    this.preserveLogicalIds = true;
+
+    // instantiate all resources as CDK L1 objects
+    for (const logicalId of Object.keys(this.template.Resources || {})) {
+      this.getOrCreateResource(logicalId);
+    }
+  }
+
+  /**
+   * Returns the low-level CfnResource from the template with the given logical ID.
+   * Any modifications performed on that resource will be reflected in the resulting CDK template.
+   *
+   * The returned object will be of the proper underlying class;
+   * you can always cast it to the correct type in your code:
+   *
+   *     // assume the template contains an AWS::S3::Bucket with logical ID 'Bucket'
+   *     const cfnBucket = cfnTemplate.getResource('Bucket') as s3.CfnBucket;
+   *     // cfnBucket is of type s3.CfnBucket
+   *
+   * If the template does not contain a resource with the given logical ID,
+   * an exception will be thrown.
+   *
+   * @param logicalId the logical ID of the resource in the CloudFormation template file
+   */
+  public getResource(logicalId: string): core.CfnResource {
+    const ret = this.resources[logicalId];
+    if (!ret) {
+      throw new Error(`Resource with logical ID '${logicalId}' was not found in the template`);
+    }
+    return ret;
+  }
+
+  /** @internal */
+  public _toCloudFormation(): object {
+    const ret: { [section: string]: any } = {};
+
+    for (const section of Object.keys(this.template)) {
+      // render all sections of the template unchanged,
+      // except Resources, which will be taken care of by the created L1s
+      if (section !== 'Resources') {
+        ret[section] = this.template[section];
+      }
+    }
+
+    return ret;
+  }
+
+  private getOrCreateResource(logicalId: string): core.CfnResource {
```

Comment:
A good name for this could be `createResourceSafe()`

Reply:
Why...? What's "safe" about it?